src/query-builder/dialects/mongo.ts

```diff
@@ -0,0 +1,301 @@
+import { QueryType } from '../../query-params'
+import { QueryBuilder } from '../../client'
+import { Query } from '../../query'
+import { MongoClient, Collection, ObjectId } from 'mongodb'
+import { AbstractDialect, ColumnDataType } from '../index'
+
+export class MongoDialect extends AbstractDialect {
+    reservedKeywords: string[] = []
+    // protected sqlFunctions: { [key: string]: (...args: string[]) => string }
+
+    getFunction(funcName: string): (...args: string[]) => string {
+        throw new Error('Method not implemented.')
+    }
+    protected addFunction(
+        funcName: string,
+        implementation: (...args: string[]) => string
+    ): void {
+        throw new Error('Method not implemented.')
+    }
+    mapDataType(dataType: ColumnDataType | string): string {
+        throw new Error('Method not implemented.')
+    }
+
+    formatSchemaAndTable(schema: string | undefined, table: string): string {
+        throw new Error('Method not implemented.')
+    }
+
+    formatFromSchemaAndTable(
+        schema: string | undefined,
+        table: string
+    ): string {
+        throw new Error('Method not implemented.')
+    }
+    select(builder: QueryBuilder, type: QueryType, query: Query): Query {
+        let selectColumns = ''
+        let fromTable = ''
+
+        const joinClauses = builder.joins?.join(' ') || ''
+
+        builder.columnsWithTable?.forEach((set, index) => {
+            // Some people who use sql might want to use *,
+            // By passing no value into projection it selects all.
+            selectColumns += set.columns
+                .filter((x) => x && x !== '*')
+                .map((columnName) => `"${columnName}": 1,`)
+            // Remove last ,
+            selectColumns = selectColumns.slice(0, -1)
+            fromTable += set.table
+        })
+
+        query.query = `db.${fromTable}.find({}, {"projection": {${selectColumns}}})`
+
+        if (joinClauses) {
+            query.query += ` ${joinClauses}`
+        }
+
+        if (builder.whereClauses?.length ?? 0 > 0) {
+            query.query += ` WHERE ${builder?.whereClauses?.join(' AND ')}`
+        }
```

Comment:
This looks very SQL-esque, this works in Mongo too?

Reply:
Left over copy/paste